compiler/rustc_mir_transform/src/inline.rs

```diff
@@ -699,7 +699,19 @@ impl<'tcx> Inliner<'tcx> {
         // Insert all of the (mapped) parts of the callee body into the caller.
         caller_body.local_decls.extend(callee_body.drain_vars_and_temps());
         caller_body.source_scopes.extend(&mut callee_body.source_scopes.drain(..));
-        caller_body.var_debug_info.append(&mut callee_body.var_debug_info);
+        if self
+            .tcx
+            .sess
+            .opts
+            .unstable_opts
+            .inline_mir_preserve_debug
+            .unwrap_or(self.tcx.sess.opts.debuginfo != DebugInfo::None)
```

Comment:
Why isn't this `== DebugInfo::Full`? Isn't this current implementation retaining variable debuginfo when compiling with `line-tables-only`?

Reply:
Before, it was always copied during inlining, so it at least shouldn't be a regression, right? And when the debuginfo is not set to generate variable debuginfo, shouldn't `var_debug_info` be empty anyway?

Reply:
> so it at least shouldn't be a regression, right?

Yeah, this *ought not* to be a regression, but I was reading this code and this condition is surprising.

> And when the debuginfo is not set to generate variable debuginfo, shouldn't `var_debug_info` be empty anyway?

When we load `optimized_mir` for the standard library, it should always have variable debuginfo.

Reply:
Oh, I see. So, I thought about familiarizing myself a bit with the debuginfo handling, I'll try to send a PR that improves this then.